File: Layout/LayoutLib/Character.cs
using System.Collections.Generic;
using System.Drawing;

namespace LayoutLib
{
    public class Character : IRect
    {
        #region IRect Impl
        public Rectangle Rect { get; }
        #endregion

        #region public properties
        public char Value => Chars[Offset];
        #endregion

        #region private properties
        private string Chars { get; }
        private int Offset { get; }
        #endregion

        #region Constructors
        public Character(string chars, int offset, Rectangle rect)
        {
            if (offset < 0 || offset >= chars.Length)
            {
                throw new ArgumentOutOfRangeException(offset, nameof(offset), 0, chars.Length - 1);
            }

            Chars  = chars;
            Offset = offset;
            Rect   = rect;
        }
        #endregion

#if DEBUG
        public override string ToString() => Value.ToString();
#endif
    }
}

File: Layout/FLayoutLib/Character.fs
namespace FLayoutLib

open System
open System.Drawing
open System.Collections.Immutable

type Character(chars: ImmutableArray<char>, offset:int, rect:Option<Rectangle>) =
    let _offset = offset
    
    member public this.Value = chars.[_offset]
    
    interface IRect with
        member this.Rect : Option<Rectangle> = rect

#if Debug
    member public this.ToString = this.Value.ToString
#endif
